src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/service/TelemetryServiceTest.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.digital.hmpps.hmppsuserpreferences.service
+
+import com.microsoft.applicationinsights.TelemetryClient
+import com.nhaarman.mockitokotlin2.verify
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+
+@ExtendWith(MockitoExtension::class)
+internal class TelemetryServiceTest {
+  @Mock
+  lateinit var telemetryClient: TelemetryClient
+
+  @InjectMocks
+  lateinit var telemetryService: TelemetryService
+
+  @Test
+  fun `Put preferences should store preferences and return DTO`() {
```

Comment:
Seems odd that there's only one test but Kotlin guarantees that all of the parameters are non-null which covers the edge cases. I think.

Reply:
yeah... seems pretty robust! Kotlin did good 😄